### PR TITLE
Issue/400 Update `building-and-running.md` document

### PR DESCRIPTION
**Pull request by t83714**
### What this PR does

Updating document `building-and-running.md` as described in issue #400 

1> included `GNU tar` & `docker` on `Prerequisites` list
2> rewrittent the content regarding starting `magda-combined-db` via `npm run dev`. `magda-combined-db` should be launched using `helm`.
3> Added `magda-authorization-api` to dependencies list of `magda-*-connector`
4> Added a link to `deploy/helm/magda/requirements.yaml`

**Todo: still need to add architecture graph from Alex's blog. Will add more commits to this PR. **

### Checklist
- [x] There are unit tests to verify my changes are correct | Unit tests aren't applicable (delete one)
- [x] I've updated CHANGES.md with what I changed.
- [x] I've linked this PR to an issue in ZenHub (or there is no issue) and dragged it to the _Pull Request_ column

### CODE REVIEW

File: doc/building-and-running.md [hunk 1]
```diff
@@ -80,32 +82,66 @@ kubectl create -f kubernetes/generated/local/
...
+A typical use case would be:
+
+1. Start `magda-combined-db` in `Minikube` using `helm:`
+
+From root level of the project directory:
+```bash
+helm install --name magda deploy/helm/magda -f deploy/helm/minikube-dev.yml --set tags.all=false --set tags.combined-db=true
+```
+
+2. Port forward database service port to localhost so that your local running program (outside the `kubenates` cluster in `minikube`) can connect to them:
```
Comment (AlexGilleran) [inline doc/building-and-running.md]: Kubernetes :)

Comment (t83714) [reply doc/building-and-running.md]: Thank you :-)

**Comment by t83714** [comment]
@AlexGilleran I've also added the architecture diagram from your blog to `What do I need to run?` section and corrected the typo in my previous commit.
Could you please have a look at the content and let me know whether it's OK? Thank you!

File: doc/building-and-running.md [hunk 2]
```diff
@@ -80,32 +82,66 @@ kubectl create -f kubernetes/generated/local/
...
+You can find more information regarding starting `magda` components via `helm` in `minikube` from [the `Running on Minikube` section below](#running-on-minikube)
+
 ## What do I need to run?
 
 Running individual components is easy enough, but how do we get a fully working system?  It is rarely necessary to run _all_ of MAGDA locally, but various components depend on other components as follows:
 
 | Component | Dependencies |
 | --------- | ------------ |
-| `magda-*-connector` | `magda-registry-api` |
+| `magda-*-connector` | `magda-registry-api`, `magda-authorization-api` |
```
Comment (AlexGilleran) [inline doc/building-and-running.md]: Actually I just turned off auth for registry-api in minikube in another PR, so we don't need this once the other PR gets merged.

File: doc/building-and-running.md [hunk 3]
```diff
@@ -116,6 +152,35 @@ Running individual components is easy enough, but how do we get a fully working
 | `magda-web-client` | `magda-web-server`, but uses API at http://magda-dev.terria.io/api if server is not running. |
 | `magda-web-server` | none, but if this is running then `magda-gateway` and its dependencies must be too or API calls will fail. |
 
+### Architecture Diagram
+
+The following `Architecture Diagram` may help you to get clearer idea which components you need to run in order to look at a particular function area:
+
+![Magda Architecture Diagram](./magda-architecture.png)
+
+The following table shows the relationship between `Madga commponents` and `Diagram elements`:
```
Comment (AlexGilleran) [inline doc/building-and-running.md]: Magda components :)

File: doc/building-and-running.md [hunk 4]
```diff
@@ -116,6 +152,35 @@ Running individual components is easy enough, but how do we get a fully working
...
+
+| Component | Diagram elements |
+| --------- | ---------------- |
+| `magda-admin-api` | `Admin API (NodeJS)` |
+| `magda-*-connector` | `Connectors` |
+| `magda-elastic-search` | `ES Clinet`, `ES Data (x2)`, `ES Master (x3)` |
```
Comment (AlexGilleran) [inline doc/building-and-running.md]: ES Client

**Comment by t83714** [comment]
@AlexGilleran Thanks a lot for pointing out the typos --- have corrected them and removed `magda-authorization-api` from dependencies list of `magda-*-connector`.

**Pull request merged**